### ISSUE TITLE
Code System and Value Set for Dose Form

### ISSUE THREAD

**Issue by costateixeira** [hl7-eu/unicom-ig#23]
We need a code system and a value set for dose form.
This should be defined in the IG so that it can be used in profiles and can be used to validate examples, and then deployed to any server

**Comment by Censacrof** [comment]
Hi, for the T6.1 we used the EDQM coding system because it was specified in the MAL, so I would suggest to use that. I have access to the EDQM APIs, and with those I have already downloaded all (or most) of the relevant valuesets. I attached to this message the files with all the EDQM valuesets for DoseForm, UnitOfPresentation, RouteOfAdministration and Container. If you need them, I also have also the full SMS valueset for Substance (which is too big to be attached here).


[container.csv](https://github.com/hl7-eu/unicom-ig/files/9858953/container.csv)
[pharmaceutical_dose_form.csv](https://github.com/hl7-eu/unicom-ig/files/9858956/pharmaceutical_dose_form.csv)
[route_of_administration.csv](https://github.com/hl7-eu/unicom-ig/files/9858959/route_of_administration.csv)
[unit_of_presentation.csv](https://github.com/hl7-eu/unicom-ig/files/9858961/unit_of_presentation.csv)

**Comment by costateixeira** [comment]
Dear Francesco

Can you please point to the EDQM API? Is there an endpoint for CSV export?
I know only the one for consulting manually.
Thank you

We can take the CSVs and convert them to FHIR CodeSystems to be able to use
them easily

Thank you

**Comment by Censacrof** [comment]
Actually, its not the API endpoint returns JSON, not CSV. I made the conversion to CSV because it was more useful for us. Anyway, you can read the [EDQM API documentation here](https://standardterms.edqm.eu/api/api_doc#!/). You need to register before being able to use it.

**Comment by rlindstrm** [comment]
I don't know what your use case is, but in real life you would probably need the following dose forms list from EDQM as well:
- combined terms
- combined pharmaceutical dose forms
- combination packs (yes, it's also dose forms, definately used for authorised dose form)

You can also download these value sets from EMA SPOR and they should provide EMA RMS code together with EDQM code.

To give EDQM codes any machine-readability and reasoning power, you also need to get the attributes/characteristics for these dose forms. I've only seen this information in pdf format (I have never tried their API), but I'm sure they have better outputs as well, because they have used this information in mapping initiatives, etc. In browser, you can see they have some good information about each dose form:
![image](https://user-images.githubusercontent.com/112903984/197836433-4b6b3495-16c6-4e57-aaf5-f9475505a3f4.png)

**Comment by rlindstrm** [comment]
There will also be EDQM-SNOMED mapping available very soon, so using SNOMED for reasoning is worth considering.

**Comment by costateixeira** [comment]
I guess this github issue just made it to the top of our agenda next
Friday 28th...?

**Comment by Censacrof** [comment]
> I don't know what your use case is, but in real life you would probably need the following dose forms list from EDQM as well:
> 
> * combined terms
> * combined pharmaceutical dose forms
> * combination packs (yes, it's also dose forms, definately used for authorised dose form)
> 
> You can also download these value sets from EMA SPOR and they should provide EMA RMS code together with EDQM code.
> 
> To give EDQM codes any machine-readability and reasoning power, you also need to get the attributes/characteristics for these dose forms. I've only seen this information in pdf format (I have never tried their API), but I'm sure they have better outputs as well, because they have used this information in mapping initiatives, etc. In browser, you can see they have some good information about each dose form: ![image](https://user-images.githubusercontent.com/112903984/197836433-4b6b3495-16c6-4e57-aaf5-f9475505a3f4.png)

Yes, I think you are right. In IDMP there are multiples kind of Doseforms, each located on different parts of the data model. For the moment we are using just Pharmaceutical Dose Form for the sake of simplicity, but we will have to also use the other ones.
